### ISSUE TITLE
UNIQUE

### ISSUE THREAD

**Issue by crysan** [realm/realm-java#967]
How to determine the class that the field name unique? Just as Primary keys?

**Comment by emanuelez** [comment]
Yes, so far that's the only way, but we're considering to support some kind of annotation for field value uniqueness

**Comment by cmelchior** [comment]
Just to be curious. Do you have a use case for this that isn't already covered by using @PrimaryKey?

**Comment by crysan** [comment]
The problem was solved. Only after the introduction of a unique field in the table, we had to alter the program logic.
There were:
        realm.beginTransaction();
        Authors author = realm.createObject(Authors.class);
        author.setUrl_base("pptr");
        realm.commitTransaction();
Now:
        Authors author = new Authors();
        author.setUrl_base("pptr");
        realm.beginTransaction();
        realm.copyToRealmOrUpdate(author);
        realm.commitTransaction();
Very unfortunate that because of one field had to rewrite the code.

**Comment by emanuelez** [comment]
That happens because you don't set your unique field. If you did you would not have to use standalone objects

**Comment by crysan** [comment]
Ok.

**Comment by mclark4386** [comment]
@emanuelez sorry I'm confused, would you mind giving a little code snippet of what you are talking about?

**Comment by crysan** [comment]
But I cited above code that used prior to the introduction to the table a unique field, and after it. Now you have to write one line of code to add more new rows to a table.

**Comment by cmelchior** [comment]
SQLite also has different modes for conflict resolution when it comes to uniqueness constraints we can consider: https://www.sqlite.org/lang_conflict.html

**Comment by tarigancana** [comment]
:+1:  waiting for this enhancement :)

**Comment by AlokBansal8** [comment]
@beeender @cmelchior Is this feature coming anytime soon?

**Comment by wyvern610** [comment]
Realm is lacking important feature. so in practical development, realm is useless.
Realm has no roadmap.so don't expect anything.

**Comment by Zhuinden** [comment]
@cmelchior now that Primary Keys also imply that the objects cannot be changed, is there a chance for a `@Unique` constraint which **doesn't** force you into the "value cannot be changed once persisted" limitation?

**Comment by merhawifissehaye** [comment]
+1

**Comment by 2mwrzesinsky** [comment]
+1

**Comment by krebse** [comment]
+1

**Comment by YaroslavYrm** [comment]
+1

**Comment by lednhatkhanh** [comment]
Any news on this, I'm still waiting for this feature

**Comment by shobhitpuri** [comment]
+1

**Comment by dtmandroid** [comment]
I want to make field 'itemName' of Table Item to be unique in Android Realm. It means 'itemName' should not contain duplicate values. Is there any other way to achieve this except Primary key.. Thanks..

**Comment by ScheNi** [comment]
+1

**Comment by Zhuinden** [comment]
Check is possible reliably in transaction if count > 0

**Comment by ViperAlpha** [comment]
+1

**Comment by comm1x** [comment]
+1

**Comment by lordplagus02** [comment]
+1

**Comment by zeallat** [comment]
+1

**Comment by hermamitr** [comment]
+1

**Comment by PiWhy** [comment]
+1

**Comment by nganthony** [comment]
+1

**Comment by rudysuharyadi** [comment]
+1, and i need it on realm-cocoa too.

**Comment by Ramzi-Alqrainy** [comment]
+1

**Comment by mblsha** [comment]
+1

**Comment by vzool** [comment]
+1

**Comment by luben93** [comment]
+1

**Comment by Zhuinden** [comment]
just for sake of completion, apparently ObjectBox 2.0 added `@Unique`: https://objectbox.io/objectbox-2-0/

**Comment by vzool** [comment]
@Zhuinden This is `ObjectBox` not **Realm**! Are they the same?

**Comment by Zhuinden** [comment]
No, they are two completely different products from two completely different companies.

To be honest, I'm not fond of ObjectBox's migration scheme, so I'm not using it; but they are working on the same domain as Realm.

**Comment by JonathanImperato** [comment]
+1

**Comment by Abdelsattar** [comment]
+1 too

**Comment by him2016** [comment]
+1

**Comment by nikitoSha** [comment]
Unique two fields - when?

**Comment by dyegos** [comment]
+1

**Comment by JohannesJoestar** [comment]
+1

**Comment by charlykno64** [comment]
+1

**Comment by maikotrindade** [comment]
+1
2020

**Comment by Romancha** [comment]
+1

**Comment by anthony-alekseiev** [comment]
+1

**Comment by lukluca** [comment]
+1

**Comment by jaredmdobson** [comment]
<img src="https://media3.giphy.com/media/xT5LMLPhW6qWTHjaWk/giphy.gif"/>

**Comment by yell66w** [comment]
+1

**Comment by ksanderer** [comment]
+1

**Comment by mancioshell** [comment]
+1
